### PR TITLE
Enrollment verification updates

### PR DESCRIPTION
**Pull request by binq**
## Summary
- Updated enrollment verification logic to account for nils we are seeing in prod. The Enrollment Logic handles determining when an enrollment period is ready to be verified by the veteran. The data supplied is run through each of the 11 different cases until one is satisfied and is returned as a period that is ready for verification. If none of the cases are satisfied, then the enrollment period is not ready to be verified. 

## Testing
- Testing has been handled on local machine manually and with the creation of RSpec tests.
- Award end Date can be NIL and the previous code did not handle for NIL. 
- This code has been tested on local machine and now handles NIL values within the enrollment logic. 
- Peer review of this code was completed as well by walking through each case with another developer to ensure each case is correct based on legacy code written in Java. 

### Test CASE EOM
image

### Test CASE  1A
image

### Test CASE 1B
image

### Test CASE 2
image

### Test CASE 3
image

### Test CASE 4
image

### Test CASE 5
image

### Test CASE 6
image

### Test CASE 7
image

### Test CASE 8
image

### Test CASE 9
image

### CODE REVIEW

**Review by wafimohamed — APPROVED** [review]

File: modules/vye/app/models/concerns/vye/needs_enrollment_verification.rb [hunk 1]
```diff
@@ -148,33 +165,38 @@ def eval_case1a
...
         number_hours: @award.number_hours,
         monthly_rate: @award.monthly_rate,
         payment_date: @award.payment_date,
         trace: :case1b
       )
+
+      true
     end
 
     def eval_case2
```
Comment (rmtolmach) [inline modules/vye/app/models/concerns/vye/needs_enrollment_verification.rb:193]: Is it possible to give these numbered cases more descriptive names? They all seem to be very similar and I'm having a hard time understanding what makes them different.

**Comment by rmtolmach** [comment]
@binq in the description can you paste the testing steps (beyond running specs) that were done to verify this is working?

File: modules/vye/app/models/concerns/vye/needs_enrollment_verification.rb (hunk 1 above)
Comment (jsimonVA) [reply modules/vye/app/models/concerns/vye/needs_enrollment_verification.rb:193]: @rmtolmach It's hard to describe them other than as "Case". We refer to this as the Enrollment Logic as a whole and each "Case" is a check that is completed to determine if the veteran has any verifications that are ready to be verified in MGIB.   In short, this logic runs through key pieces of data such as the award's start date, end date, and date last certified for that individual veteran and determines if the current data supplied is ready to be verified. There are several different times a veteran's enrollment verification is ready to be verified and each "Case" is one of those reasons. Please let me know if you would like to meet up and talk this through.

**Comment by rmtolmach** [comment]
@binq Please include screenshots next time of your testing.

Is it possible to test your changes on [your Review Instance](http://7e5a11d5a340e34fe7f918cb0e0fa2cd.review.vetsgov-internal/)?

File: modules/vye/app/models/concerns/vye/needs_enrollment_verification.rb (hunk 1 above)
Comment (rmtolmach) [reply modules/vye/app/models/concerns/vye/needs_enrollment_verification.rb:193]: Having an understanding of the code isn't just for ME, it's a maintainability issue. Contracts don't stay around forever, so I want to make sure the team maintaining this code down the road understands what's happening in this file. There's no way these case names could be more descriptive than numbering them? Is there something that makes each case unique? If they need to have the numbering convention, could a comment be added to each method? For example, the comments in this file explaining the acronyms are helpful:
```ruby
   # date_last_certified is before award_begin_date
    def dlc_before_abd?
```

**Comment by jsimonVA** [comment]
> @binq Please include screenshots next time of your testing.
> 
> Is it possible to test your changes on [your Review Instance](http://7e5a11d5a340e34fe7f918cb0e0fa2cd.review.vetsgov-internal/)?

@rmtolmach I just ran all "Cases" and attached screenshots.

**Pull request merged**